File: cpp/dnn_control/fullstatesensorsimulator.cpp
#include "fullstatesensorsimulator.h"

FullStateSensorSimulator::FullStateSensorSimulator(const Asteroid &asteroid, const double &sensor_noise) : SensorSimulator(asteroid), normal_distribution_(boost::mt19937(time(0)),boost::normal_distribution<>(0.0, sensor_noise))
{
    dimensions_ = 7;
}

FullStateSensorSimulator::~FullStateSensorSimulator()
{

}

void FullStateSensorSimulator::Simulate(const State &state, const Vector3D &perturbations_acceleration, const double &time, SensorData &sensor_data)
{
    for (int i = 0; i < dimensions_; ++i) {
        sensor_data[i] = state[i] + state[i] * normal_distribution_();
    }
}

File: python/dnn_control/fullstatesensorsimulator.py
class FullStateSensorSimulator:

    def __init__(self, asteroid, sensor_noise):
        self._asteroid = asteroid
        self._sensor_noise = sensor_noise

    # Generates (simulates) sensor data based on the current spacecraft state "state" and time "time"
    def simulate(self, state, perturbations_acceleration, time):
        return state